ADALiOS/ADALiOS/ADTokenCacheStoring.h

```diff
@@ -66,4 +66,27 @@
                    userId:(NSString *)userId
                     error:(ADAuthenticationError * __autoreleasing *)error;
 
+/*! Mark a cache item as dead instead of removing it.
+ @param item: the item being marked as dead. Key will be extracted from the item.
+ @param userId: The user for which the item will be removed. Can be nil, in which case items for all users with
+ the specified key will be removed.
+ @param error: in case of an error, if this parameter is not nil, it will be filled with
+ the error details. */
+- (void)tombstoneItem:(ADTokenCacheStoreItem *)item
```

Comment:
This is a public interface, so anything you add to it also becomes public. Tombstoning is more a private implementation detail that we should minimize the exposure of.